audio-worklet/bitcrusher-test.html

```diff
@@ -15,6 +15,172 @@
 <!DOCTYPE html>
   <head>
     <script src="src/bitcrusher-script-processor.js"></script> 
-    <script src="test/bitcrusher-test.js"></script> 
+    <script>
+    	testSampleReduction();
+    	testBitcrusherInContext();
+    	testMultipleChannels();
+    	testIllegalInput();
+
+    	/**
+    	 * Populate array with 0:99 and verify the bitcrusher holds an array
+    	 * where the number of consecutive equal numbers is controlled by the 
+    	 * variable factor.
+    	 */
+    	function testSampleReduction() {
+    	  let context = new OfflineAudioContext(2, 44100, 44100);
+
+    	  for (let reduction = 1; reduction < 10; reduction++) {
+    	    const bufferSize = 5;
+    	    let referenceBuffer = new Float32Array(bufferSize);
+    	    let outputBuffer = new Float32Array(bufferSize);
+    	    let bitcrusher =
+    	        new Bitcrusher(context, {bitDepth: 24, reduction: reduction});
+
+    	    for (let i = 0; i < bufferSize; i++) {
+    	      referenceBuffer[i] = i;
+    	    }
+
+    	    bitcrusher.processBuffer_(referenceBuffer, outputBuffer);
+
+    	    // Verify computed values match expected values.
+    	    outputBuffer.map((data, index) => {
+    	      let expected = Math.floor(index / reduction) * reduction;
+    	      console.assert(
+    	          data === expected, 
+    	          'computed ' + data + ' but expected ' + expected);
+    	    });
+    	  }
+    	}
+
+    	/**
+    	 * Create one oscillator, and connect it to a bitcrusher with variables 
+    	 * set to avoid any effect and a delay node which compensates for the 
+    	 * script processor latency. Then verify that the samples are nearly
+    	 * identical.
+    	 */
+    	function testBitcrusherInContext() { 
```

Comment:
I have tests which interact directly with the bitcrusher functions and others which interact with those functions through AudioContext. I should express this? testBitcrusherInAudioContext?